src/main.rs

```diff
@@ -1421,11 +1422,11 @@ fn validate_signed_delegate_action(
                 }) => {
                     debug!("method_name: {:?}", method_name);
                     debug!("deposit: {:?}", deposit);
-                    if !is_whitelisted_da_receiver {
+                    if state.config.use_whitelisted_contracts && !is_whitelisted_da_receiver {
```

Comment:
@anthony-near possible logic bug, some of the tests were getting caught up here, despite whitelisting being disabled.

Reply:
This line should only get hit if `state.config.use_exchange == true`. Were some tests failing on other test cases where `use_exchange == false` due to this line?

Reply:
Yes exactly, some tests were failing out early because `use_whitelisted_***` was `false` while `use_exchange` was `true`

Reply:
And you're sure it was this line and not another line? The rust compiler + failed test messages are good, but they're not always accurate in what caused the actual failure. There's also:
```
    if !state.config.use_fastauth_features
        && !is_whitelisted_da_receiver
        && state.config.use_whitelisted_contracts
```
and
```
    if !is_whitelisted_da_receiver
        && state.config.use_fastauth_features
        && state.config.use_whitelisted_contracts
```

which could have caused the tests to fail

Reply:
Here's what failed when we don't have that line: https://github.com/near/pagoda-relayer-rs/pull/85  

Perhaps it's how I've set up the tests themselves? Maybe I made a false assumption somewhere.